src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java

```diff
@@ -158,23 +164,26 @@ void merge(ProcessConfig processConfig, ZonedDateTime dateTime, String date, Str
 
                 if (processConfig.isRunBalancesAdjustment()) {
                     // balances adjustment on the merge network
-                    balancesAdjustmentService.doBalance(networkUuids);
-
-                    LOGGER.info("Merge {} of process {}: balance adjustment complete", date, processConfig.getProcess());
+                    return balancesAdjustmentService.doBalance(networkUuids).flatMap(res -> {
+                        LOGGER.info("Merge {} of process {}: balance adjustment complete", date, processConfig.getProcess());
 
-                    // TODO check balance adjustment status
-                    mergeEventService.addMergeEvent(processConfig.getProcess(), dateTime, MergeStatus.BALANCE_ADJUSTMENT_SUCCEED);
+                        // TODO check balance adjustment status
+                        mergeEventService.addMergeEvent(processConfig.getProcess(), dateTime, MergeStatus.BALANCE_ADJUSTMENT_SUCCEED);
+                        return Mono.empty();
+                    });
                 } else {
                     // load flow on the merged network
-                    loadFlowService.run(networkUuids);
-
-                    LOGGER.info("Merge {} of process {}: loadflow complete", date, processConfig.getProcess());
+                    return loadFlowService.run(networkUuids).flatMap(res -> {
```

Comment:
.then instead of flatMap?

Reply:
same response

Reply:
You're right, can't use then(); Instead, use doOnSuccess ?